hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -83,12 +86,34 @@ protected AbstractRecordStore(MapContainer mapContainer, int partitionId) {
         this.mapStoreContext = mapContainer.getMapStoreContext();
         this.mapDataStore = mapStoreContext.getMapStoreManager().getMapDataStore(name, partitionId);
         this.lockStore = createLockStore();
-        Collection<RecordStoreMutationObserver<Record>> mutationObservers = mapServiceContext
-                .createRecordStoreMutationObservers(getName(), partitionId);
-        this.mutationObserver = new CompositeRecordStoreMutationObserver<>(mutationObservers);
+        this.mutationObserver = new CompositeMutationObserver<>();
         this.storeAdapter = new RecordStoreAdapter(this);
     }
 
+    @Override
+    public void addMutationObserver(MutationObserver<Record> observer) {
+        mutationObserver.add(observer);
+    }
+
+    @Override
+    public void init() {
+        this.storage = createStorage(recordFactory, inMemoryFormat);
+        initMutationObservers();
+    }
+
+    private void initMutationObservers() {
```

Comment:
We have two private `initMutationObservers()` methods. One for `AbstractRecordStore` and one for the enterprise sub-class, so we need this `inMemoryFormat != NATIVE` condition here. I'd consider making `initMutationObservers()` protected with adding a `getJsonMetadataMutationObserver()` method called from `AbstractRecordStore#initMutationObservers()`. WDYT?

Reply:
done.